tpcc/check.go

```diff
@@ -3,128 +3,270 @@ package main
 import (
 	"database/sql"
 	"fmt"
+
+	"time"
+
+	"github.com/pkg/errors"
 )
 
-func forEachWarehouseAndDistrict(fn func(wID, dID int) error) error {
-	for wID := 0; wID < *warehouses; wID++ {
-		for dID := 1; dID < 10; dID++ {
-			if err := fn(wID, dID); err != nil {
-				return err
-			}
+func check3321(db *sql.DB) error {
+	start := time.Now()
+
+	var rowsLeft, rowsRight *sql.Rows
+	var err error
+	if rowsLeft, err = db.Query("SELECT w_ytd FROM warehouse ORDER BY w_id"); err != nil {
+		return err
+	}
+
+	if rowsRight, err = db.Query("SELECT SUM(d_ytd) FROM district GROUP BY d_w_id ORDER BY d_w_id"); err != nil {
+		return err
+	}
+
+	var left, right float64
+	var i int
+	for ; rowsLeft.Next() && rowsRight.Next(); i++ {
```

Comment:
A client side merge join? Couldn't you combine the queries above into a FULL OUTER JOIN and then check whether the results had missing rows on either the left or right side?